### DISCUSSION TITLE
tree-sitter-c: identifying how field_identifier is appearing under pointer_declarator

### DISCUSSION THREAD

**Discussion by berchn** [tree-sitter/tree-sitter#1022] [Q&A] [answered]
With the C parser, ` struct B { A* x; }; ` produces the following:

```
translation_unit [0, 0] - [1, 0])
  struct_specifier [0, 0] - [0, 18])
    name: type_identifier [0, 7] - [0, 8])
    body: field_declaration_list [0, 9] - [0, 18])
      field_declaration [0, 11] - [0, 16])
        type: type_identifier [0, 11] - [0, 12])
        declarator: pointer_declarator [0, 12] - [0, 15])
          declarator: field_identifier [0, 14] - [0, 15])
```
I'm having trouble identifying where the `field_identifier` is coming from with this parse tree as there doesn't appear to be a direct path from `pointer_declarator` to `field_identifier`.

**Comment by theHamsta** [comment, marked as answer]
This is because the C grammar uses  a pointer_field_declarator node. The alias let's it appeat as a field_declarator. It's a lie!

https://github.com/tree-sitter/tree-sitter-c/blob/5aa0bbbfc41868a3727b7a89a90e9f52e0964b2b/grammar.js#L294
